src/app/legacy/containers/SocialEmbed/sourceHelpers.js

```diff
@@ -35,9 +35,16 @@ export const getProviderFromSource = source => {
 export const getIdFromSource = source => {
   const NO_ID = '';
   const provider = getProviderFromSource(source);
-  if (provider === PROVIDERS.TWITTER) {
-    const match = source.match(/\/status\/([0-9]+)/);
-    return match ? match[1] : NO_ID;
+  if (
+    provider === PROVIDERS.TWITTER ||
+    provider === PROVIDERS.YOUTUBE ||
+    provider === PROVIDERS.INSTAGRAM
+  ) {
+    const twitterId = source.match(/\/status\/([0-9]+)/);
+    const youtubeId = source.match(/\/watch\?v=([0-9 A-Z a-z _-]+)/);
+    const instagramId = source.match(/\/p\/([0-9 A-Z a-z]+)/);
```

Comment:
hmm. Something can be done here, I just don't have the time or headspace to see what it is.

Reply:
Here, I could have gone for an extended hard-to-read regex, I preferred splitting it for readability.
And I am sure there are some tricks that can be performed with arrays... But again, I don't think they will be as readable.

Reply:
I can use an object that holds the correct regex and call it with the provider e.g.
I think something like this won't harm readability:
```

const regex = {
  twitter: /\/status\/([0-9]+)/, 
  youtube: /\/watch\?v=([0-9 A-Z a-z _-]+)/, 
  Instagram: /\/p\/([0-9 A-Z a-z]+)/
}
const id = source.match(regex[provider]);
```

Reply:
think I prefer this. You?